### PR TITLE
Minor typo fixing and code refactoring

### PR DESCRIPTION
**Pull request by tensorush**
Hi! Great work with this TUI library!

I'd like to contribute these fixes:
- Added standard `.gitattributes` file for Zig projects.
- Reworked `build.zig` a little, hopefully it's a bit clearer. Also, now `zig build` will run all steps.
- `outer: while` in examples was redundant since there's only one loop to break from. `switch` expressions don't allow breaking from them, so breaking is only for loops, i.e. `while` and `for`.
- When returning a struct instance from a function, the compiler infers the return type from function signature, so instead of `return MyType{...};` , it's more idiomatic to write `return .{...};`.
- Logging adds a new line by default, so you don't usually need to write `\n` like here: `log.debug("event: {}\r\n", .{event});`.

Please tell me if something doesn't make sense, I'll be sure to fix it!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,28 +16,28 @@ Contributions are welcome.
...
+| Color Mode Updates (DEC 2031)  |   ✅   | planned  |     ❌     |
+| Images (full/space)            |   ✅   | planned  |     ✅     |
+| Images (half block)            |   ✅   | planned  |     ✅     |
+| Images (quadrant)              |   ✅   | planned  |     ✅     |
+| Images (sextant)               |   ❌   |    ❌     |     ✅     |
+| Images (sixel)                 |   ✅   | debating |     ✅     |
+| Images (kitty)                 |   ✅   |    ✅     |     ✅     |
+| Images (iterm2)                |   ❌   |    ❌     |     ✅     |
+| Video                          |   ❌   |    ❌     |     ✅     |
+| Dank                           |  🆗   |    🆗    |     ✅     |
```
Comment (rockorager) [inline README.md:40]: What formatter did you use? I use prettier for markdown files and it did the formatting of this table automatically.

File: src/cell.zig
Comment (rockorager) [inline src/cell.zig:1]: This file should be renamed `Cell.zig` I believe

**Comment by rockorager** [comment]
Thanks! A couple comments left inline. I appreciate it! Always like to have it be more idiomatic :)

The `\r\n` in the logs was for when the terminal is in raw mode (IE if you are logging but not redirecting stderr), you'll need the `\r` to get everything back to column 0. But generally logging should be redirected so I'm good with removing those.

File: README.md (hunk 1 above)
Comment (tensorush) [reply README.md:40]: I used this one:
https://marketplace.visualstudio.com/items?itemName=darkriszty.markdown-table-prettify

Should I revert that formatting?

File: src/cell.zig
Comment (tensorush) [reply src/cell.zig:1]: Oh, yeah, will do

File: README.md (hunk 1 above)
Comment (rockorager) [reply README.md:40]: Yeah, let's stick with what prettier outputs. Thanks

**Review by rockorager — COMMENTED** [review]
Huh....seems like the file rename isn't here?

**Comment by tensorush** [comment]
Yeah, had to do a `git mv` there. Should be there now.

**Pull request merged**